packages/style-engine/class-wp-style-engine.php

```diff
@@ -186,41 +189,82 @@ protected static function get_classnames( $style_value, $style_definition ) {
 	 *
 	 * @param array         $style_value      A single raw style value from the generate() $block_styles array.
 	 * @param array<string> $style_definition A single style definition from BLOCK_STYLE_DEFINITIONS_METADATA.
+	 * @param boolean       $should_return_css_vars Whether to try build and return CSS var values.
 	 *
 	 * @return array        An array of CSS rules.
 	 */
-	protected static function get_css( $style_value, $style_definition ) {
-		// Low-specificity check to see if the value is a CSS preset.
+	protected static function get_css( $style_value, $style_definition, $should_return_css_vars ) {
+		$rules = array();
+
+		if ( ! $style_value ) {
+			return $rules;
+		}
+
+		// Before default processing, style definitions could define a callable `value_func` to generate custom CSS rules at this point.
+		$style_property = $style_definition['property_key'];
+
+		// Build CSS var values from var:? values, e.g, `var(--wp--css--rule-slug )`
+		// Check if the value is a CSS preset and there's a corresponding css_var pattern in the style definition.
 		if ( is_string( $style_value ) && strpos( $style_value, 'var:' ) !== false ) {
-			return array();
+			if ( $should_return_css_vars && $style_definition['css_vars'] ) {
```

Comment:
This doesn't cause any issues just yet, but in case there is no `css_vars` property, to avoid an `Undefined index` notice, shall we add an `isset` or `empty` check? (I think avoiding the `isset` check for `property_key` is probably fine, given that it's a required prop for each of the block style definitions, and they're hard-coded in the same file)

```suggestion
			if ( $should_return_css_vars && ! empty( $style_definition['css_vars'] ) ) {
```